addons/website_crm_partner_assign_commissions/models/account_invoice.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+from odoo.tools import float_compare
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    reseller_id = fields.Many2one('res.partner', 'Reseller', help='This field is used to track the reseller in order to generate commisions')
+    purchase_order_id = fields.Many2one('purchase.order', 'Reseller Purchase Order')
+
+    def make_commissions(self):
+        """
+            Adds a line to a purchase order for reseller_id containing the due commissions for the invoice
+            The line is only added the first time we call this method (called on reconcile)
+        """
+        self.ensure_one()
+
+        if not self.reseller_id:
+            return
+
+        subtotals = []
+        for commission in self.env['res.partner.commission'].search([('grade_id', '=', self.reseller_id.grade_id.id)]):
+            lines = self.invoice_line_ids.filtered(lambda line: line.product_id.product_tmpl_id.id in commission.product_ids.ids)
+            subtotals.append(sum(line.price_subtotal * commission.percentage / 100 for line in lines))
+        total = sum(subtotals)
+
+        if not self.purchase_order_id and total:
```

Comment:
If a purchase order is already linked the code below isn't execute and doesn't modify the PO.